bentoml/_internal/frameworks/tensorflow_v2.py

```diff
@@ -78,50 +72,17 @@ def _clean_name(name: str) -> str:  # pragma: no cover
     return re.sub(r"\W|^(?=\d)-", "_", name)
 
 
-@inject
-def load(
-    bento_tag: t.Union[str, Tag],
+def _load_model(
+    bentomodel: BentoModel,
     tags: t.Optional[t.List[str]] = None,
-    options: t.Optional["tf_ext.SaveOptions"] = None,
     load_as_hub_module: t.Optional[bool] = None,
-    model_store: "ModelStore" = Provide[BentoMLContainer.model_store],
 ) -> t.Union["tf_ext.AutoTrackable", "tf_ext.Module", "HubModule", "KerasLayer"]:
-    """
-    Load a model from BentoML local modelstore with given name.
 
-    Args:
-        bento_tag (:code:`Union[str, Tag]`):
-            Tag of a saved model in BentoML local modelstore.
-        tags (:code:`str`, `optional`, defaults to `None`):
-            A set of strings specifying the graph variant to use, if loading from a v1 module.
-        options (:code:`tensorflow.saved_model.SaveOptions`, `optional`, default to :code:`None`):
-            :code:`tensorflow.saved_model.LoadOptions` object that specifies options for loading. This
-            argument can only be used from TensorFlow 2.3 onwards.
-        load_as_hub_module (`bool`, `optional`, default to :code:`True`):
-            Load the given weight that is saved from tfhub as either `hub.KerasLayer` or `hub.Module`.
-            The latter only applies for TF1.
-        model_store (:mod:`~bentoml._internal.models.store.ModelStore`, default to :mod:`BentoMLContainer.model_store`):
-            BentoML modelstore, provided by DI Container.
-
-    Returns:
-        :obj:`SavedModel`: an instance of :obj:`SavedModel` format from BentoML modelstore.
-
-    Examples:
-
-    .. code-block:: python
-
-        import bentoml
-
-        # load a model back into memory
-        model = bentoml.tensorflow.load("my_tensorflow_model")
-
-    """  # noqa: LN001
-    model = model_store.get(bento_tag)
-    if model.info.module not in (MODULE_NAME, __name__):
+    if bentomodel.info.module not in (MODULE_NAME, __name__):
         raise BentoMLException(
```

Comment:
Should we take this time to rework exceptions and start having more specific exception types?